includes/class-wc-payments-account.php

```diff
@@ -838,15 +848,104 @@ public function maybe_redirect_from_onboarding_page(): bool {
 		}
 
 		// Don't redirect merchants that have no Stripe account connected.
-		if ( ! $this->is_stripe_connected() ) {
+		if ( ! $this->is_stripe_connected() || ! $this->is_details_submitted() ) {
 			return false;
 		}
 
-		$this->redirect_service->redirect_to_overview_page( 'WCPAY_ONBOARDING_FLOW' );
+		$this->redirect_service->redirect_to_overview_page( WC_Payments_Onboarding_Service::FROM_ONBOARDING_WIZARD );
 
 		return true;
 	}
 
+	/**
+	 * Redirects connect page (payments/connect) to the overview page for stores that
+	 * have a working Jetpack connection and a connected Stripe account.
+	 *
+	 * Note: Connect _page_ links are not the same as connect links.
+	 *       Connect links are used to start/re-start/continue the onboarding flow and they are independent of
+	 *       the WP dashboard page (based solely on request params).
+	 *
+	 * IMPORTANT: The logic should be kept in sync with the one in maybe_redirect_from_overview_page to avoid loops.
```

Comment:
I'm guessing you probably already covered this, but if not, is there a unit test we can put in place that will validate this?

Reply:
Good call, Daniel! I've added [the anti-looping tests](https://github.com/Automattic/woocommerce-payments/pull/9105/commits/862d8a1658180dc63e6cf185e6f88e1fcb0ab715).